src/osd/ReplicatedPG.cc

```diff
@@ -5977,6 +5979,23 @@ int ReplicatedPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
       }
       break;
 
+    case CEPH_OSD_OP_ASSERT_INTERVAL:
+      ++ctx->num_read;
+      {
+	epoch_t epoch = op.assert_interval.epoch;
+	epoch_t pg_epoch = get_epoch();
```

Comment:
the osdmap's epoch might be greater than the one of a scrub happened in current interval. i'd suggest use `info.history.same_interval_since` instead.